src/ln/channelmanager.rs

```diff
@@ -1465,23 +1465,55 @@ impl ChannelManager {
 	/// still-available channels.
 	fn fail_htlc_backwards_internal(&self, mut channel_state_lock: MutexGuard<ChannelHolder>, source: HTLCSource, payment_hash: &[u8; 32], onion_error: HTLCFailReason) {
 		match source {
-			HTLCSource::OutboundRoute { .. } => {
+			HTLCSource::OutboundRoute { ref route, .. } => {
 				mem::drop(channel_state_lock);
-				if let &HTLCFailReason::ErrorPacket { ref err } = &onion_error {
-					let (channel_update, payment_retryable) = self.process_onion_failure(&source, err.data.clone());
-					if let Some(update) = channel_update {
+				match &onion_error {
+					&HTLCFailReason::ErrorPacket { ref err } => {
+#[cfg(test)]
```

Comment:
Hmm does using conditional compilation speed-up so much the tests ? Really personal, but don't find that it eases code readability..

Reply:
needed not to generate additional compile warning message.

Reply:
maybe #[allow(dead_code)] ? but okay